executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Chain.hs

```diff
@@ -113,105 +110,99 @@ instance ( HashAlgorithm hashAlgo
-                                   , openingSlot = openingSlot
-                                   , closingSlot = slot
-                                   , vpDuration = vpDuration
-                                   , vpStatus = VPClosed
-                                   }
-                              else -- VP should remain open
-                                vp
-                        )
-                        open
-
-              -- extract all closed VPs from open Map
-              newClosedVPs =
-                Map.filter (\VotingPeriod {vpStatus} ->
-                                vpStatus == VPClosed
-                           )
-                           updatedOpen
-            in
-              -- insert new closed VPs into closed Map and remove closed VPs from open Map
-              ( Map.difference open newClosedVPs
-              , Map.union newClosedVPs closed
-              )
-
-      -- NOTE: the TRANSACTIONS transition corresponds to the BODY transition in
-      -- Byron and Shelley rules.
-      transactionsSt'@Transaction.St { Transaction.openVotingPeriods = ovp'
-                                        -- ovp' = returned state updated by IDEATION
-                                     , Transaction.utxoSt = _
-                                     , Transaction.updateSt = _
-                                     } <-
-        trans @(TRANSACTIONS hashAlgo)
-          $ TRC ( Transaction.Env slot closedVotingPeriods' upE utxoE
-                  -- pass the updated openVotingPeriods state
-                , Transaction.St openVotingPeriods' utxoSt updateSt
-                , transactions
+
+      -- First a HEAD transition in order to update the state
+      headerSt'@Header.St { Header.currentSlot = slot'
+                          , Header.hupdateSt
+                              = Hupdate.St { Hupdate.wrsips = wrsips'
+                                           , Hupdate.asips = asips'
+                                           }
+                          }  <-
+        trans @(HEADER hashAlgo)
+          $ TRC ( headerEnv
+                , headerSt
+                , header
                 )
-      pure $! St { currentSlot = slot
-                 , openVotingPeriods = ovp' -- This state has been further updated
-                                            -- by the IDEATON STS
-                 , closedVotingPeriods = closedVotingPeriods'
-                 , transactionsSt = transactionsSt'
+
+      -- Second a BODY transition with the updated state from header
+      bodySt'<- trans @(BODY hashAlgo)
+                  $ TRC ( Body.Env
+                           { Body.currentSlot = slot'
+                           , Body.asips = asips'
+                           , Body.transactionEnv
+                           }
+                        , Body.St
+                            { Body.wrsips = wrsips'
+                            , Body.transactionSt
```

Comment:
I think there might be a duplication of `wrsips` in the transactions state and in the update state.